python/ray/data/_internal/logical/operators.py

```diff
@@ -0,0 +1,61 @@
+import sys
+from typing import Any, Dict, Iterable, Optional, Union
+from ray.data._internal.compute import ComputeStrategy
+from ray.data._internal.lazy_block_list import LazyBlockList
+
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.data.block import BatchUDF
+from ray.data.datasource.datasource import Datasource
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+class Read(LogicalOperator):
+    """Logical operator for read."""
+
+    def __init__(
+        self,
+        blocks: LazyBlockList,
+        datasource: Datasource,
```

Comment:
How about depending on Reader instead of the entire Datasource class to narrow the dependency?

Reply:
SG, updated.